src/sync/sync-utils.ts

```diff
@@ -85,6 +85,7 @@ const resetTargetedTasks = async (subscription: Subscription, syncType?: SyncTyp
 		}
 		updateRepoSyncTasks[`${task}Status`] = null;
 	});
+	updateRepoSyncTasks[`failedCode`] = null;
```

Comment:
So when the `targetTasks` is empty, this won't be cleared?

Reply:
I think it will be ok if we can pass the tasks from the front end rather than moving to a seperate function. Now we've two functions `resetTargetedTasks` and `resetFailedCode` which are doing the pretty similar tasks but on different fields which makes the code complex to understand.  It just a thought!